src/lib/stylejson.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+  validate as validateStyleJSON,
+  StyleSpecification as StyleJSON,
+  SourceSpecification,
+} from '@maplibre/maplibre-gl-style-spec'
+
+import { encodeBase32, generateId, hash } from './utils'
+
+type OfflineSource = SourceSpecification & {
+  tilesetId: string
+}
+
+type OfflineStyle = StyleJSON & {
+  id: string
```

Comment:
yeah that makes sense. does that change in the case of the `GET /styles`endpoint, where we're retrieving all of them?

Reply:
Updated so that there's no standalone type representing the idea of offline (since it was getting confusing). In the case of `POST /styles/`, the response will include an id field i.e. `StyleJSON & IdResource`